chapter3/challenges/program/lib/password_manager_2.rb

```diff
@@ -1,2 +1,54 @@
 class PasswordManager2
-end
\ No newline at end of file
+    def initialize
+        #Initializes three arrays, password manager being the main one, and two other arrays for checking / ease of ordering.
+        @password_manager = Array.new()
+        @password_array = Array.new()
+        @service_array = Array.new()
+    end
+        def add(service, password)
+           new_entry = {
+               "service" => service,
+               "password" => password,
+               "added" => DateTime.now()
+           }
+   
```

Comment:
I like that you have made this, since it makes it clearer what is being added @password_manager later. But wouldn't it be even clearer if this code was in the only block it was going to be used? (The final else block). Your program makes this with every add request, even if it isn't going to use it.